src/sys/resource.rs

```diff
@@ -1,68 +1,69 @@
-
-#[cfg(any(target_os = "linux",
-          target_os = "openbsd",
-          target_os = "netbsd",
-          target_os = "bitrig"))]
-pub use libc::{RLIM_SAVED_CUR, RLIM_SAVED_MAX};
+use libc::{self, c_int, rlimit, RLIM_INFINITY};
+pub use libc::rlim_t;
 
 use {Errno, Result};
 
-#[derive(Clone, Copy, PartialEq, Eq, Debug)]
-#[repr(i32)]
-pub enum Resource {
-    // POSIX
-    RLIMIT_CORE = libc::RLIMIT_CORE,
-    RLIMIT_CPU = libc::RLIMIT_CPU,
-    RLIMIT_DATA = libc::RLIMIT_DATA,
-    RLIMIT_FSIZE = libc::RLIMIT_FSIZE,
-    RLIMIT_NOFILE = libc::RLIMIT_NOFILE,
-    RLIMIT_STACK = libc::RLIMIT_STACK,
-    RLIMIT_AS = libc::RLIMIT_AS,
-    // BSDs and Linux
-    #[cfg(all(unix, not(target_os = "solaris")))]
-    RLIMIT_MEMLOCK = libc::RLIMIT_MEMLOCK,
-    #[cfg(all(unix, not(target_os = "solaris")))]
-    RLIMIT_NPROC = libc::RLIMIT_NPROC,
-    #[cfg(all(unix, not(target_os = "solaris")))]
-    RLIMIT_RSS = libc::RLIMIT_RSS,
-    // Linux-only
-    #[cfg(any(target_os = "linux", target_os = "android"))]
-    RLIMIT_LOCKS = libc::RLIMIT_LOCKS,
-    #[cfg(any(target_os = "linux", target_os = "android"))]
-    RLIMIT_MSGQUEUE = libc::RLIMIT_MSGQUEUE,
-    #[cfg(any(target_os = "linux", target_os = "android"))]
-    RLIMIT_NICE = libc::RLIMIT_NICE,
-    #[cfg(any(target_os = "linux", target_os = "android"))]
-    RLIMIT_RTPRIO = libc::RLIMIT_RTPRIO,
-    #[cfg(any(target_os = "linux", target_os = "android"))]
-    RLIMIT_RTTIME = libc::RLIMIT_RTTIME,
-    #[cfg(any(target_os = "linux", target_os = "android"))]
-    RLIMIT_SIGPENDING = libc::RLIMIT_SIGPENDING,
-}
+libc_enum!{
+    #[repr(i32)]
+    pub enum Resource {
+        // POSIX
+        RLIMIT_AS,
+        RLIMIT_CORE,
+        RLIMIT_CPU,
+        RLIMIT_DATA,
+        RLIMIT_FSIZE,
+        RLIMIT_NOFILE,
+        RLIMIT_STACK,
+
+        // BSDs and Linux
+        #[cfg(any(target_os = "android", target_os = "freebsd", target_os = "linux", target_os = "openbsd"))]
+        RLIMIT_MEMLOCK,
+        #[cfg(any(target_os = "android", target_os = "freebsd", target_os = "linux", target_os = "openbsd"))]
+        RLIMIT_NPROC,
+        #[cfg(any(target_os = "android", target_os = "freebsd", target_os = "linux", target_os = "openbsd"))]
+        RLIMIT_RSS,
+
+        // Android and Linux only
+        #[cfg(any(target_os = "android", target_os = "linux"))]
+        RLIMIT_LOCKS,
+        #[cfg(any(target_os = "android", target_os = "linux"))]
+        RLIMIT_MSGQUEUE,
+        #[cfg(any(target_os = "android", target_os = "linux"))]
+        RLIMIT_NICE,
+        #[cfg(any(target_os = "android", target_os = "linux"))]
+        RLIMIT_RTPRIO,
+        #[cfg(any(target_os = "android", target_os = "linux"))]
+        RLIMIT_RTTIME,
+        #[cfg(any(target_os = "android", target_os = "linux"))]
+        RLIMIT_SIGPENDING,
 
-#[inline]
-fn rlim_to_option(rlim: rlim_t) -> Option<rlim_t> {
-    match rlim {
-        RLIM_INFINITY => None,
-        rlim => Some(rlim),
+        // Non-Linux
```

Comment:
The comment makes it sound like these symbols are available everywhere except Linux.

Reply:
Changed to `// Available on some BSD`